search-app/src/cmr/search/data/query_to_elastic.clj

```diff
@@ -421,17 +427,22 @@
         specified-sort (q2e/sort-keys->elastic-sort concept-type sort-keys)
         default-sort (q2e/sort-keys->elastic-sort concept-type (q/default-sort-keys concept-type))
         sub-sort-fields (if (:all-revisions? query)
-                          collection-all-revision-sub-sort-fields
+                          (all-revision-sub-sort-fields :collection)
                           collection-latest-sub-sort-fields)
         ;; We want the specified sort then to sub-sort by the score.
         ;; Only if neither is present should it then go to the default sort.
         specified-score-combined (seq (concat specified-sort score-sort-order))]
     (concat (or specified-score-combined default-sort) sub-sort-fields)))
 
-(defmethod q2e/concept-type->sub-sort-fields :variable
-  [_]
-  [{(q2e/query-field->elastic-field :name :variable) {:order "asc"}}
-   {(q2e/query-field->elastic-field :provider :variable) {:order "asc"}}])
+(defmethod q2e/query->sort-params :variable
+  [query]
+  (let [{:keys [concept-type sort-keys]} query
+        specified-sort (q2e/sort-keys->elastic-sort concept-type sort-keys)
+        default-sort (q2e/sort-keys->elastic-sort concept-type (q/default-sort-keys concept-type))
+        sub-sort-fields (if (:all-revisions? query)
```

Comment:
Nitpick: I think we should avoid having conditions inside `let` blocks. To me, the block below is a little easier to read and refactor in the future if needed

```
(concat
  (or (q2e/sort-keys->elastic-sort concept-type sort-keys)
      (q2e/sort-keys->elastic-sort concept-type (q/default-sort-keys concept-type))
  (if (:all-revisions? query)
    (all-revision-sub-sort-fields :variable)
    variable-latest-sub-sort-fields)))

Reply:
I am fine with `if` within let block. In this case, it is basically setting a intermediate variable and the variable name (`sub-sort-fields` makes the code more readable). The combined way is harder to read in my opinion.